File: Euler-Particles/src/shape/Circle.java
package shape;

import processing.core.PApplet;
import processing.core.PVector;

public class Circle extends RadialShape {

    public Circle(PApplet p) {
        super(p);
    }

    public void display() {
        PApplet p = getPApplet();
        PVector position = getPosition();
        p.noStroke();
        p.fill(getColor());
        p.pushMatrix();
        p.translate(position.x, position.y);
        p.ellipse(0, 0, getRadius() * 2, getRadius() * 2);
        p.popMatrix();
    }
}

File: docs/Euler-Particles/shape/circle.js
class Circle extends RadialShape {

    constructor() {
        super();
    }

    display() {
        push();
        noStroke();
        fill(this.color.getColor());
        translate(this.position.x, this.position.y);
        ellipse(0, 0, this.radius * 2, this.radius * 2);
        pop();
    }
}
